repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,254 @@
+
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
+
+    try:
+        updater = Updater(
+            metadata_dir=metadata_dir,
+            metadata_base_url=metadata_url,
+            target_base_url=artifacts_url,
+            target_dir=download_dir,
+        )
+        updater.refresh()
+
+        info = updater.get_targetinfo(artifact_name)
+
+        if info is None:
+            console.print(f"Artifact {artifact_name} not found")
+            return False
+
+        path = updater.find_cached_target(info)
+        if path:
+            console.print(f"Artifact is available in {path}")
+            return True
+
+        path = updater.download_target(info)
+        console.print(f"Artifact downloaded and available in {path}")
+
+    except (OSError, RepositoryError, DownloadError) as e:
+        console.print(f"Failed to download artifact {artifact_name}: {e}")
+        return False
+
+    return True
+
+
+@artifact.command()
+@click.option(
+    "-r",
+    "--root",
+    help="A metadata URL to the initial trusted root or a local file.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-m",
+    "--metadata-url",
+    help="TUF Metadata repository URL.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A hash prefix.",
+    is_flag=True,
+    required=None,
+)
+@click.option(
+    "-P",
+    "--directory-prefix",
+    help="A prefix for the download dir.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.argument("artifact_name")
+@click.pass_context
+def download(
+    context: Context,
+    root: str,
+    metadata_url: str,
+    artifacts_url: str,
+    hash_prefix: bool,
+    directory_prefix: str,
+    artifact_name,
+) -> None:
+    """
+    Downloads artifacts to the TUF metadata.
+    """
+
+    rstuf_config = context.obj.get("settings")
+
+    if context.obj.get("config") and os.path.isfile(context.obj.get("config")):
+        repository = rstuf_config.get("CURRENT_REPOSITORY")
+        # Overwriting the config if a flag is passed
+        # or using it as default instead
+        if not repository:
+            console.print("Please specify current repository")
+            sys.exit()
+        if not rstuf_config.get("REPOSITORIES"):
+            console.print("No reposotiroes listed in the config file")
+            sys.exit()
+        if not rstuf_config["REPOSITORIES"].get(repository):
+            console.print(
+                f"Repository {repository} is missing in the "
+                "configuration file"
+            )
+            sys.exit()
+        if not artifacts_url:
+            artifacts_url = rstuf_config["REPOSITORIES"][repository].get(
+                "artifact_base_url"
```

Comment:
This attribute you want to get is not the same as the one you define in #459.
In #459 the name is `artifact_url`.
I think it's best if we name this `artifacts_base_url` meaning this is not a base url for 1 artifact but many.

Reply:
I don't mind unifying, even though `artifact_url` is for flag simplicity, while `artifact_base_url` is for config clarity. We support both use-cases, but they differ. I agree that `artifacts_base_url` is good for the config file, but it's improper for a flag, as we don't support downloading multiple artifacts at once.

This is why IMO it's best to have both `artifact_url` (flags only) and `artifacts_base_url` (config file). If we choose one, `artifact_url` is best for the command line, even though it's not clear enough for the config file.

Reply:
Okay, I understand your idea now.
Let's leave it like that.